### PR TITLE
First batch of new Pester tests for debugger functionality

### PR DESCRIPTION
**Pull request by KirkMunro**
# PR Summary

- updated HelpersDebugger module to support more debugging scenarios with PowerShell
- added tests for `BreakpointUpdated` and `DebuggerStop` handlers when working with `LineBreakpoint`, `VariableBreakpoint`, and `CommandBreakpoint` objects
- added debugger SDK tests

## PR Context

Bring old unit tests that were written in .NET forward as Pester tests in PowerShell.

## PR Checklist

- [X] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [X] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [X] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [X] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [X] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [X] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

File: test/powershell/Language/Scripting/Debugging/DebuggerCommand.Tests.ps1 [hunk 1]
```diff
@@ -214,14 +214,13 @@ Describe 'Basic debugger command tests' -tag 'CI' {
         }
```
Comment (PoshChan) [inline test/powershell/Language/Scripting/Debugging/DebuggerCommand.Tests.ps1]: @KirkMunro, your last commit had 1 failures in `PowerShell-CI-linux`
Basic debugger command tests.Callstack (k, Get-PSCallStack) command should show the current call stack.Should only have CallStackFrame output from the callstack command
```powershell
Expected the value to have type [System.Management.Automation.CallStackFrame] or any of its subtypes, but got $null with type $null.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Language/Scripting/Debugging/DebuggerCommand.Tests.ps1: line 217
217:             $result['k'].Output | Should -BeOfType System.Management.Automation.CallStackFrame
```

**Comment by SSJ9899TDylan** [comment]
Close my ass

File: test/powershell/Language/Scripting/Debugging/DebuggerCommand.Tests.ps1 (hunk 1 above)
Comment (PoshChan) [inline test/powershell/Language/Scripting/Debugging/DebuggerCommand.Tests.ps1]: @KirkMunro, your last commit had 1 failures in `PowerShell-CI-macos`
Basic debugger command tests.Callstack (k, Get-PSCallStack) command should show the current call stack.Should only have CallStackFrame output from the callstack command
```powershell
Expected the value to have type [System.Management.Automation.CallStackFrame] or any of its subtypes, but got $null with type $null.
at <ScriptBlock>, /Users/runner/runners/2.160.1/work/1/s/test/powershell/Language/Scripting/Debugging/DebuggerCommand.Tests.ps1: line 217
217:             $result['k'].Output | Should -BeOfType System.Management.Automation.CallStackFrame
```

Comment (PoshChan) [inline test/powershell/Language/Scripting/Debugging/DebuggerCommand.Tests.ps1]: @KirkMunro, your last commit had 1 failures in `PowerShell-CI-windows`
Basic debugger command tests.Callstack (k, Get-PSCallStack) command should show the current call stack.Should only have CallStackFrame output from the callstack command
```powershell
Expected the value to have type [System.Management.Automation.CallStackFrame] or any of its subtypes, but got $null with type $null.
at <ScriptBlock>, D:\a\1\s\test\powershell\Language\Scripting\Debugging\DebuggerCommand.Tests.ps1: line 217
217:             $result['k'].Output | Should -BeOfType System.Management.Automation.CallStackFrame
```

**Review by PaulHigin — APPROVED** [review]
Just one minor comment.  Otherwise LGTM.

File: test/powershell/Language/Scripting/Debugging/Debugging.Tests.ps1 [hunk 2]
```diff
@@ -51,6 +51,174 @@ Describe 'Basic debugger tests' -tag 'CI' {
...
+<# 11 #>     [CmdletBinding(SupportsShouldProcess=$true)]
+<# 12 #>     param()
+<# 13 #>     'Line 13'
+<# 14 #>     'Line 14'
+<# 15 #>     'Line 15'
+<# 16 #> }
+<# 17 #>
+<# 18 #> Test-PSRemoveBreakpoint -WhatIf
+'@
+        $testScriptPath = [System.IO.Path]::Combine([System.IO.Path]::GetTempPath(), 'BreakpointTestScript.ps1')
```
[the remark this answers is not in the document]
Comment (KirkMunro) [reply test/powershell/Language/Scripting/Debugging/Debugging.Tests.ps1]: No, it just wasn't top of mind (old habits die hard). I'll update this test to use test drive instead.

File: test/powershell/Language/Scripting/Debugging/Debugging.Tests.ps1 [hunk 3]
```diff
@@ -51,6 +51,174 @@ Describe 'Basic debugger tests' -tag 'CI' {
...
+                #       with the debugger here, which scripts that run unattended won't do.
+                BreakpointLine        = 10
+                BreakpointColumnStart = 43
+                BreakpointColumnEnd   = 44
+            }
+        )
+    }
+
+    AfterAll {
+        [System.IO.File]::Delete($testScriptPath)
```
Comment (adityapatwardhan) [inline test/powershell/Language/Scripting/Debugging/Debugging.Tests.ps1]: This is unnecessary as the content of the test drive will be deleted.

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]
Just one small change. Rest looks good.

**Pull request closed without merging**